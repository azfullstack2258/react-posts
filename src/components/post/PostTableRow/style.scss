.post-table__row {
  display: flex;
  background: #343a40;

  svg path {
    fill: #343a40;
  }

  &:nth-child(even) {
    background: #3e444a;

    svg path {
      fill: #3e444a;
    }
  }

  &-cell {
    color: white;
    padding: 20px 10px;

    img {
      width: 80px;
      height: 80px;
    }

    &:not(.content) {
      border-right: 1px solid #464d55;
    }

    &.id {
      min-width: 30px;

    }

    &.user {
      min-width: 120px;
      max-width: 120px;

      .user-image {
        margin-top: 4px;
        padding: 10px;
        border: 1px solid #434a51;
        width: 100px;
        height: 100px;

        img {
          width: 100px;
          height: 100px;
        }

        div {
          width: 100px;
          height: 100px;
          border: 1px solid #c0c0c0;
        }
      }
    }

    &.social_network,
    &.property,
    &.post_date,
    &.post_media {
      min-width: 100px;
      max-width: 100px;
    }

    &.post_content {
      flex-grow: 1;
    }

    &.social_network {
      display: flex;
      justify-content: center;

      .social-network {
        width: 70px;
        height: 70px;
        border-radius: 50%;
        color: transparent;
        font-size: 40px;
        display: flex;
        align-items: center;
        justify-content: center;

        &.facebook {
          background: #3c5997;
        }
        &.twitter {
          background: #03aced;
        }
        &.instagram {
          background: #3f739a;
        }
        &.share {
          background: #03bf02;
        }
      }
    }

    &.post_date {
      div {
        line-height: 25px;
      }
    }

    &.post_media {
      img {
        width: 90px;
        height: 90px;
      }
      div {
        width: 100px;
        height: 100px;
        border: 1px solid #c0c0c0;
      }
    }
  }
}