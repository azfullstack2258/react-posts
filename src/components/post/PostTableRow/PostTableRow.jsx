import React, { useMemo } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import moment from 'moment';

import './style.scss';

const socialNetworkIcons = {
  facebook: ['fab', 'facebook-f'],
  twitter: ['fab', 'twitter'],
  instagram: ['fab', 'instagram'],
  share: 'share-alt'
};
const socialNetworkNames = ['facebook', 'twitter', 'instagram'];

const PostTableRow = ({ data }) => {
  const headers = useMemo(() => [
    {
      key: 'id',
    },
    {
      key: 'property',
      renderer: (item) => <div>{item ? item.name : 'Unknown'}</div>
    },
    {
      key: 'user',
      renderer: (item) => (
        <>
          <div>{item ? item.name : 'Unknown'}</div>
          <div className="user-image">
            {item
              ? <img src={item.profile_picture} alt="user_img" />
              : <div />
            }
          </div>
        </>
      )
    },
    {
      key: 'social_network',
      renderer: (name) => {
        let icon = 'share-alt';
        let socialNetwork = 'share';
  
        if (name) {
          socialNetwork = socialNetworkNames.find(sn => name.includes(sn)) || 'share';
        }
        icon = socialNetworkIcons[socialNetwork];

        return (
          <div className={`social-network ${socialNetwork}`}><FontAwesomeIcon icon={icon} /></div>
        );
      }
    },
    {
      key: 'post_date',
      renderer: (date) => date && date !== 'null' ? (
        <>
          <div>{moment(date).format('dddd')}</div>
          <div>{moment(date).format('MMM DD')}</div>
          <div>{moment(date).format('yyyy')}</div>
        </>
      ) : 'N/A'
    },
    {
      key: 'post_media',
      renderer: (image) => (
        <>
          {image && image !== 'null' ? <img src={image} alt="post_img" /> : <div />}
        </>
      )
    },
    {
      key: 'post_content',
      renderer: (content) => (
        <div
          dangerouslySetInnerHTML={
            { __html:
                (content && content.replace(/@([a-z\d_]+)/ig, '<span class="mention">@$1</span>')) || ''
            }
          }
        />
      )
    }
  ], []);

  return (
    <div className="post-table__row">
      {data ? headers.map(({ key, renderer }) => (
        <div key={key} className={`post-table__row-cell ${key}`}>
          {typeof renderer === 'function' ? renderer(data[key]) : data[key]}
        </div>
      )) : ''}
    </div>
  );
};

export default PostTableRow;
