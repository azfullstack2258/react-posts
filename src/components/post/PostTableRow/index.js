import PostTableRow from './PostTableRow';

export default PostTableRow;
