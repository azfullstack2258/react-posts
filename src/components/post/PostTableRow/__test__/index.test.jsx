import React from 'react';
import { render } from '@testing-library/react';

import PostTableRow from '../PostTableRow';

describe('<PostTableRow />', () => {
  it('should render a prop', () => {
    const { container } = render(
      <PostTableRow />
    );

    expect(container).toMatchSnapshot();
  })
});
