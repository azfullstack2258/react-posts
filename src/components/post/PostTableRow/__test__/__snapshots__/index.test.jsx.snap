// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTableRow /> should render a prop 1`] = `
<div>
  <div
    class="post-table__row"
  />
</div>
`;
